lib/ansible/modules/cloud/amazon/ec2_eni.py

```diff
@@ -574,35 +795,28 @@ def main():
             purge_secondary_private_ip_addresses=dict(default=False, type='bool'),
             secondary_private_ip_address_count=dict(default=None, type='int'),
             allow_reassignment=dict(default=False, type='bool'),
-            attached=dict(default=None, type='bool')
+            attached=dict(default=None, type='bool'),
+            name=dict(default=None, type='str'),
+            tags=dict(default={}, type='dict'),
+            purge_tags=dict(default=False, type='bool')
         )
     )
 
-    module = AnsibleModule(argument_spec=argument_spec,
-                           mutually_exclusive=[
-                               ['secondary_private_ip_addresses', 'secondary_private_ip_address_count']
-                           ],
-                           required_if=([
-                               ('state', 'absent', ['eni_id']),
-                               ('attached', True, ['instance_id']),
-                               ('purge_secondary_private_ip_addresses', True, ['secondary_private_ip_addresses'])
-                           ])
-                           )
-
-    if not HAS_BOTO:
-        module.fail_json(msg='boto required for this module')
-
-    region, ec2_url, aws_connect_params = get_aws_connection_info(module)
-
-    if region:
-        try:
-            connection = connect_to_aws(boto.ec2, region, **aws_connect_params)
-            vpc_connection = connect_to_aws(boto.vpc, region, **aws_connect_params)
-        except (boto.exception.NoAuthHandlerFound, AnsibleAWSError) as e:
-            module.fail_json(msg=str(e))
-    else:
-        module.fail_json(msg="region must be specified")
+    module = AnsibleAWSModule(
+        argument_spec=argument_spec,
+        mutually_exclusive=[
+            ['secondary_private_ip_addresses', 'secondary_private_ip_address_count']
+        ],
+        required_if=([
+            ('attached', True, ['instance_id']),
```

Comment:
The state: absent qualifier was dropped.  Is that ok?

Reply:
Yes.  Otherwise you can only delete an eni by eni_id.  I dropped it to allows delete of the eni by (name and subnet_id), (subnet_id and private_ip_address), or (instance_id and device_index).

Reply:
Leaving this conversation open - its worth getting feedback from others on this point.